lib/src/array.rs

```diff
@@ -184,23 +186,18 @@ impl YrsArray {
         arr.remove_range(tx, index, len)
     }
 
-    pub(crate) fn observe(&self, delegate: Box<dyn YrsArrayObservationDelegate>) -> u32 {
-        let subscription_id = self
+    pub(crate) fn observe(&self, delegate: Box<dyn YrsArrayObservationDelegate>) -> Arc<YSubscription> {
```

Comment:
I am not sure about this Arc<YSubscription> return type. Again, I am not a Rust developer so I am not sure about the UniFFI generated code would not compile because it expected a type like this.

Reply:
@Horusiath what do you think about this? Do you think this is safe?